src/cascadia/TerminalApp/lib/Pane.LayoutSizeNode.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "Pane.h"
+
+Pane::LayoutSizeNode::LayoutSizeNode(const float minSize) :
+    size{ minSize },
+    isMinimumSize{ true },
+    firstChild{ nullptr },
+    secondChild{ nullptr },
+    nextFirstChild{ nullptr },
+    nextSecondChild{ nullptr }
+{
+}
+
+Pane::LayoutSizeNode::LayoutSizeNode(const LayoutSizeNode& other) :
+    size{ other.size },
+    isMinimumSize{ other.isMinimumSize },
+    firstChild{ other.firstChild ? std::make_unique<LayoutSizeNode>(*other.firstChild) : nullptr },
+    secondChild{ other.secondChild ? std::make_unique<LayoutSizeNode>(*other.secondChild) : nullptr },
+    nextFirstChild{ other.nextFirstChild ? std::make_unique<LayoutSizeNode>(*other.nextFirstChild) : nullptr },
+    nextSecondChild{ other.nextSecondChild ? std::make_unique<LayoutSizeNode>(*other.nextSecondChild) : nullptr }
+{
+}
+
+// Method Description:
+// - Makes sure that this node and all its descendants equal the supplied node.
+//   This may be more efficient that copy construction since it will reuse its
+//   allocated children.
+// Arguments:
+// - other: Node to take the values from.
+// Return Value:
+// - itself
+Pane::LayoutSizeNode& Pane::LayoutSizeNode::operator=(const LayoutSizeNode& other)
+{
+    size = other.size;
+    isMinimumSize = other.isMinimumSize;
+
+    _AssignChildNode(firstChild, other.firstChild.get());

```

Comment:
Sorry, I don't understand what's wrong here. I only pop the underlying pointer to pass it to `_AssignChildNode`, which just reads the pointed object and copies it (copies object, not the pointer). It could receive it by reference as well, except it has to be nullable. Or it could receive reference to `unique_ptr` but that's anti pattern in such situations (not to force implementation). And I don't see any two `unique_ptr`s to hold the same value.